src/stores/slices/networkSlice.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * Network slice for Zustand store
+ *
+ * This slice manages network connectivity state using NetInfo:
+ * - Online/offline status
+ * - Connection type (wifi, cellular, etc.)
+ * - Server reachability
+ */
+
+import NetInfo, { NetInfoState } from "@react-native-community/netinfo";
+import { logger } from "@/lib/logger";
+import type { SliceCreator } from "@/types/store";
+
+// Create cached sublogger for this slice
+const log = logger.forTag("NetworkSlice");
+
+/**
+ * Network slice state interface - scoped under 'network'
+ */
+export interface NetworkSliceState {
+  network: {
+    /** Whether the device is connected to a network */
+    isConnected: boolean;
+    /** Whether the device can reach the internet */
+    isInternetReachable: boolean | null;
+    /** Type of network connection (wifi, cellular, etc.) */
+    connectionType: string | null;
+    /** Whether the slice has been initialized */
+    initialized: boolean;
+  };
+}
+
+/**
+ * Network slice actions interface
+ */
+export interface NetworkSliceActions {
+  /** Initialize the slice and start listening to network changes */
+  initializeNetwork: () => void;
+  /** Update network state (called by NetInfo listener) */
+  _updateNetworkState: (state: NetInfoState) => void;
+  /** Reset the slice to initial state */
+  resetNetwork: () => void;
+}
+
+/**
+ * Combined Network slice interface
+ */
+export interface NetworkSlice extends NetworkSliceState, NetworkSliceActions {}
+
+/**
+ * Initial state
+ */
+const initialState: NetworkSliceState = {
+  network: {
+    isConnected: false,
```

Comment:
`isConnected` should check to see if we are actually able to talk to the ABS server